api/src/services/files.ts

```diff
@@ -112,6 +115,27 @@ export class FilesService extends ItemsService {
 			}
 		}
 
+		if (['video/mpeg', 'audio/mpeg', 'audio/mp3', 'video/mp4', 'audio/ogg', 'video/ogg'].includes(payload.type)) {
+			const ffprobeStatic = require('@ffprobe-installer/ffprobe');
+			const buffer = await storage.disk(data.storage).getBuffer(payload.filename_disk);
+			const { path, cleanup } = await file();
+			fs.appendFile(path, buffer.content);
```

Comment:
This'll read the whole file into memory, which might cause problems on high-usage and/or low-memory environments. Could we save the file to tmp using streams instead? (or read it directly from the stream).

From a quick google, I found another project that passes a stream to execa as input for ffprobe, maybe that could be the solution? https://github.com/caffco/get-audio-duration/blob/master/src/index.ts

Reply:
Yeah, you're right I changed it and it's much faster, but now the duration only works for video files. I'll try to get it back working but if anyone knows anything about the issue please let me know.

Reply:
It seems like it's not possible for audio files using stdin https://trac.ffmpeg.org/ticket/4358

Maybe we could have two versions, one for video files using streams and one for audio files using tmp files?

Reply:
We could save the file to a temp location in a stream first, and then run ffmpeg. As long as we don't read the full file into Node's memory, we should be ok 👍🏻

Reply:
The current version is now creating the temp file using streams and seems to be working for audio and videos.